tools/rosbag/src/rosbag/bag.py

```diff
@@ -1725,7 +1736,7 @@ def __init__(self, bag):
     def start_reading(self):
         raise NotImplementedError()
 
-    def read_messages(self, topics, start_time, end_time, connection_filter, raw):
+    def read_messages(self, topics, start_time, end_time, connection_filter, raw, return_connection_header):
```

Comment:
There is no reason to break the API signature here. The new argument should be made optional. See first commit on #1473.